store/kv/kv_store.go

```diff
@@ -133,12 +137,18 @@ func (tx *KeyValueTx) UpdateRange(ctx context.Context, table []byte, lKey Key, r
 }
 
 type KeyValueIterator struct {
+	ctx context.Context
 	baseIterator
 	err error
 }
 
 func (i *KeyValueIterator) Next(value *KeyValue) bool {
```

Comment:
Just to separate the logic you can rename original method to nextLow and call it from new Next().

Reply:
Thanks, will do that!